src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.ThreadPoolValueTaskSource.cs

```diff
@@ -102,12 +103,12 @@ private void ExecuteInternal()
                                 RandomAccess.WriteAtOffset(_fileHandle, _singleSegment.Span, _fileOffset);
                                 break;
                             case Operation.ReadScatter:
-                                Debug.Assert(_readScatterBuffers != null);
-                                result = RandomAccess.ReadScatterAtOffset(_fileHandle, _readScatterBuffers, _fileOffset);
+                                Debug.Assert(_buffers != null);
+                                result = RandomAccess.ReadScatterAtOffset(_fileHandle, Unsafe.As<IReadOnlyList<Memory<byte>>>(_buffers), _fileOffset);
```

Comment:
It is not ok to use `Unsafe.As` just because of you can. In fact, we are working on eliminating Unsafe.As and similar uses to make the runtime core more secure by construction.

Reply:
Clearly, one cannot use it just like that. The reasoning was that the setting of the field is made only in one of two scatter/gather methods and this is aligned with setting the `_operation`. In my opinion it's similar to the case in `ValueTask` which also casts As according to the "type" held in another filed.

Reply:
Maybe you could make a regular cast; the always-true type check will be dwarfed by the cost of the subsequent I/O operation.

Reply:
Replaced by the regular